Widgets/PrivacyIndicator/Indicator.swift

```diff
@@ -0,0 +1,77 @@
+//import Foundation
+import UIKit
+
+public enum PrivacyIndicator {
+    
+public class Widget: UIView {
+    public var onTapBlock: (() -> Void)? {
+        didSet {
+            self.button.isHidden = onTapBlock == nil
+        }
+    }
+    private var canvas = PrivacyIndicator.CanvasView()
+    private var button = PrivacyIndicator.ButtonView()
+
+    override public func didMoveToSuperview() {
+        super.didMoveToSuperview()
+        self.clipsToBounds = false
+        self.addSubview(self.canvas)
+        self.addSubview(self.button) // should be added last, see hitTest
+        self.button.addTarget(
+            self,
+            action: #selector(self.didPressButton(_:)),
+            for: .touchUpInside)
+    }
+    
+    override public func hitTest(_ point: CGPoint, with event: UIEvent?) -> UIView? {
+        if clipsToBounds || isHidden || alpha == 0 { return nil }
+        for subview in subviews.reversed() {
+            let newPoint = subview.convert(point, from: self)
+            let view = subview.hitTest(newPoint, with: event)
+            if view != nil { return view }
+        }
+        return nil
+    }
+} // end class Widget
+} // end namespace PrivacyIndicator
+
+public extension PrivacyIndicator {
+    typealias Segment = (UIColor, Int)
+}
+
+public extension PrivacyIndicator.Widget {
+    func update(
+        arcs: [PrivacyIndicator.Segment],
+        strike: PrivacyIndicator.Segment?
+    ) {
+        self.canvas.update(arcs: arcs, strike: strike)
+    }
+}
+
+fileprivate extension PrivacyIndicator.Widget {
+    @objc
+    func didPressButton(_ button: UIButton) {
+        self.onTapBlock?()
+    }
+}
+
+
+// TODO fix preview
+#if canImport(SwiftUI) && DEBUG
```

Comment:
As we are supporting iOS started 11.4 we couldn't use SwiftUI.

Reply:
Yeap, agree
Somehow I managed to see preview (and keep version 11.4 :)
Seems for preview version does not make some sense 
<img width="1629" alt="Screenshot 2020-02-03 at 13 37 52" src="https://user-images.githubusercontent.com/38907423/73653786-627cd200-468a-11ea-8233-2a0bd3691d7a.png">